internal/provider/waypoint/data_source_waypoint_application_template.go

```diff
@@ -0,0 +1,227 @@
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var _ datasource.DataSource = &DataSourceApplicationTemplate{}
+var _ datasource.DataSourceWithConfigValidators = &DataSourceApplicationTemplate{}
+
+func (d DataSourceApplicationTemplate) ConfigValidators(ctx context.Context) []datasource.ConfigValidator {
+	return []datasource.ConfigValidator{
+		datasourcevalidator.Conflicting(
+			path.MatchRoot("name"),
+			path.MatchRoot("id"),
+		),
+	}
+}
+
+type DataSourceApplicationTemplate struct {
+	client *clients.Client
+}
+
+type DataSourceApplicationTemplateModel struct {
+	ID                     types.String `tfsdk:"id"`
+	Name                   types.String `tfsdk:"name"`
+	ProjectID              types.String `tfsdk:"project_id"`
+	OrgID                  types.String `tfsdk:"organization_id"`
+	Summary                types.String `tfsdk:"summary"`
+	Labels                 types.List   `tfsdk:"labels"`
+	Description            types.String `tfsdk:"description"`
+	ReadmeMarkdownTemplate types.String `tfsdk:"readme_markdown_template"`
+
+	TerraformCloudWorkspace *tfcWorkspace    `tfsdk:"terraform_cloud_workspace_details"`
+	TerraformNoCodeModule   *tfcNoCodeModule `tfsdk:"terraform_no_code_module"`
+}
+
+func NewApplicationTemplateDataSource() datasource.DataSource {
+	return &DataSourceApplicationTemplate{}
+}
+
+func (d *DataSourceApplicationTemplate) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_waypoint_application_template"
+}
+
+func (d *DataSourceApplicationTemplate) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The Waypoint Template data source retrieves information on a given Application Template.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Optional:    true,
+				Description: "The ID of the Application Template.",
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Application Template.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Waypoint Application Template is located.",
+				Computed:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Waypoint Application Template is located.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"summary": schema.StringAttribute{
+				Description: "A brief description of the template, up to 110 characters",
+				Computed:    true,
```

Comment:
Should this be required instead of computed? It is required by Waypoint in the validator.

Reply:
For anyone else reading, `summary` should actually be computed, but so should be everything else here except for `name`, `project_id`, and `id`. So a small change was required but not the one indicated in the comment above.